csrc/scheduler/heuristic.h

```diff
@@ -25,11 +25,20 @@ class HeuristicParams : public PolymorphicBase {
     return "Undefined Heuristic Params";
   }
 
-  virtual size_t hash() const = 0;
-
-  virtual bool sameAs(const std::shared_ptr<HeuristicParams>& other) const = 0;
+  virtual size_t hash() const {
+    return 0;
+  };
+
+  virtual bool sameAs(const std::shared_ptr<HeuristicParams>& other) const {
+    if (!other->isA<HeuristicParams>()) {
```

Comment:
You might want `other->isStrictlyA<HeuristicParams>()` to do a strict check. Otherwise this will be false unless other is `nullptr`.